webpack/assets/javascripts/react_app/components/ChartBox/ConnectedChartBox.js

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { CHARTS_DATA } from './ChartBoxConstants';
+import { translate as __ } from '../../common/I18n';
+import ChartBox from './';
+
+import { get } from '../../redux/API';
+import { selectChart } from './ChartBoxSelectors';
+
+const ConnectedChartBox = ({
+  className,
+  config,
+  noDataMsg,
+  errorText,
+  type,
+  chart,
+  tip,
+}) => {
+  const key = `${CHARTS_DATA}_${chart.id}`;
+  useEffect(() => {
+    dispatch(get({ key, url: chart.url }));
+  }, [key, chart, dispatch]);
+  const chartData = useSelector(state => selectChart(state, key));
+  const dispatch = useDispatch();
+  return (
+    <ChartBox
+      className={className}
+      config={config}
+      noDataMsg={noDataMsg}
+      errorText={errorText || (chartData.error && chartData.error.message)}
+      type={type}
+      tip={tip}
+      chart={chartData}
+      id={chart.id}
+      status={chartData.status}
+      title={chart.title}
+      search={chart.search}
```

Comment:
can we use the selectors in a better way to return each of the chart property separately?

Reply:
I deconstructed and split the selectors, is that what you meant?